### PR TITLE
f-registration@v0.36.0

### PR DESCRIPTION
**Pull request by luke-andrew-parry**
This PR adds handling for `403` responses when an account is created. This can happen when the authentication attempt on the newly created account gets either blocked by ravelin or a Recaptcha is served. In these instances, we need to navigate the user to the login page to manually log in.

### CODE REVIEW

**Review by adamashton — DISMISSED** [review]

File: packages/f-registration/src/components/Registration.vue [hunk 1]
```diff
@@ -416,16 +416,19 @@ export default {
                     thrownErrors = error.response.data.errors;
                 }
 
+                this.$emit(EventNames.CreateAccountFailure, thrownErrors);
+
                 if (Array.isArray(thrownErrors)) {
                     if (thrownErrors.some(thrownError => thrownError.errorCode === '409')) {
                         this.shouldShowEmailAlreadyExistsError = true;
+                    } else if (thrownErrors.some(thrownError => thrownError.errorCode === '403')) {
+                        window.location.href = this.copy.navLinks.login.url;
```
Comment (tommcclean) [inline packages/f-registration/src/components/Registration.vue]: Just a question: Do we want to do the navigation here or in the parent component, I ask because I think when an account is registered successfully; the website routes the user back to the homepage (not the component), does this mean the user doesn't seem a message saying why? (is that desirable)

Comment (luke-andrew-parry) [reply packages/f-registration/src/components/Registration.vue]: > Just a question: Do we want to do the navigation here or in the parent component, I ask because I think when an account is registered successfully; the website routes the user back to the homepage (not the component),

Good point. I can change this to emit an event instead and hook into it on the parent component to redirect.

> does this mean the user doesn't seem a message saying why? (is that desirable)

This is how it behaves currently. It's quite an edge case but I agree it's not a very nice experience for the user.

**Review by smarkey — DISMISSED** [review]
LGTM from test perspective

**Review by adamashton — DISMISSED** [review]

File: packages/f-registration/src/components/Registration.vue [hunk 2]
```diff
@@ -415,17 +415,24 @@ export default {
                 if (error && error.response && error.response.data && error.response.data.errors) {
                     thrownErrors = error.response.data.errors;
                 }
+                let shouldEmitCreateAccountFailure = true;
 
                 if (Array.isArray(thrownErrors)) {
                     if (thrownErrors.some(thrownError => thrownError.errorCode === '409')) {
                         this.shouldShowEmailAlreadyExistsError = true;
+                    } else if (thrownErrors.some(thrownError => thrownError.errorCode === '403')) {
```
Comment (cmyhill-justeat) [inline packages/f-registration/src/components/Registration.vue:428]: Would extending this to a switch statement be more elegant?

**Review by tommcclean — DISMISSED** [review]

File: packages/f-registration/src/components/tests/Registration.test.js [hunk 3]
```diff
@@ -133,6 +133,26 @@ describe('Registration', () => {
                 }
             });
 
+            it('should emit login blocked event when service responds with a 403', async () => {
+                // Arrange
+                const err = { response: { data: { faultId: '123', traceId: '123', errors: [{ description: 'Not authorized.', errorCode: '403' }] } } };
```
Comment (xander-marjoram) [inline packages/f-registration/src/components/tests/Registration.test.js]: Nit: Isn't 403 Forbidden and 401 is Unauthorized?

**Review by xander-marjoram — COMMENTED** [review]
Code looks good, just one small comment

File: packages/f-registration/src/components/tests/Registration.test.js (hunk 3 above)
Comment (luke-andrew-parry) [reply packages/f-registration/src/components/tests/Registration.test.js]: Yes, that should be forbidden.

File: packages/f-registration/src/components/Registration.vue (hunk 2 above)
Comment (luke-andrew-parry) [reply packages/f-registration/src/components/Registration.vue:428]: I tried it but to be honest with the amount of indenting I think it looks more verbose and noisy 😅 

```                    
if (thrownErrors.some(thrownError => thrownError.errorCode === '409')) {
    this.shouldShowEmailAlreadyExistsError = true;
} else if (thrownErrors.some(thrownError => thrownError.errorCode === '403')) {                        
    this.$emit(EventNames.LoginBlocked)
    shouldEmitCreateAccountFailure = false;
} else {
    this.genericErrorMessage = thrownErrors[0].description || 'Something went wrong, please try again later';
}
```
vs
```
thrownErrors.forEach(thrownError => {
    switch (thrownError.error) {
        case '409':
            this.shouldShowEmailAlreadyExistsError = true;
            break;
        case '403':
            this.$emit(EventNames.LoginBlocked);
            shouldEmitCreateAccountFailure = false;
            break;
        default:
            this.genericErrorMessage = thrownErrors[0].description || 'Something went wrong, please try again later';
    }
});
```

Comment (cmyhill-justeat) [reply packages/f-registration/src/components/Registration.vue:428]: Yeah not so keen on iterating the entire collection would rather reduce it first.  Yeah looks like switch it not the best solution here

**Review by cmyhill-justeat — DISMISSED** [review]

**Comment by luke-andrew-parry** [comment]
@cmyhill-justeat @tommcclean @smarkey @adamashton Would you mind putting your reviews back on, please? Had to push up a small test change.

**Review by cmyhill-justeat — DISMISSED** [review]

**Review by adamashton — DISMISSED** [review]

**Review by xander-marjoram — APPROVED** [review]

**Review by smarkey — APPROVED** [review]
